src/box/txn.c

```diff
@@ -1183,8 +1189,20 @@ txn_commit_impl(struct txn *txn, enum txn_commit_wait_mode wait_mode)
 		txn->fiber = NULL;
 		return 0;
 	}
-	while (!req->is_complete)
-		fiber_yield();
+	while (!req->is_complete) {
+		if (box_begin_timeout_new_meaning) {
```

Comment:
Nit: no sense in doing that in `while (!req->is_complete)` cycle. We cannot retry it anyway

Reply:
AFAIU it's possible to observe spurious wakeup while using fiber (I don't see any execution sequence to reproduce, but apparently it's possible). Therefore, in all such places, we wrap `fiber_yield` in a loop with a condition check.
This is also recommended even when using `fiber_cond_wait` and `fiber_cond_wait_timeout`:
https://github.com/tarantool/tarantool/blob/222770d5242e23b91ba5bcf3d9ef7c919a04af40/src/lib/core/fiber_cond.h#L110-L130